mergebot_poller.py

```diff
@@ -0,0 +1,47 @@
+"""TODO(jasonkuster): DO NOT SUBMIT without one-line documentation for mergebot_poller.
+
+TODO(jasonkuster): DO NOT SUBMIT without a detailed description of mergebot_poller.
+"""
+
+import mergebot
+import Queue
+
+GITHUB_SECRET = '../../github_auth/apache-beam.secret'
+
+class MergebotPoller:
+  scm_type = ""
+  repository = ""
+  branch = ""
+  scm_pollers = { "github": poll_github }
+  scm_mergers = { "github": mergebot.merge_git }
+  work_queue = Queue.Queue()
+
+
+  def poll(self, config):
+    print('Starting merge poller for {}.'.format(config['name']))
+    self.scm_type = config['scm_type']
+    self.repository = config['repository']
+    self.branch = config['default_branch']
+    scm_pollers[scm_type](config)
+
+
+  def poll_github(self, config):
```

Comment:
Comment does not agree with the actual sleep below. Also not that since polling may take longer than the 1m period, you may need to account for that.